boards/xtensa/waveshare_esp32s3_touch_lcd/waveshare_esp32s3_touch_lcd.dts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024 Joel Guittet
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <espressif/esp32s3/esp32s3_r2.dtsi>
+#include "waveshare_esp32s3_touch_lcd-pinctrl.dtsi"
+#include <zephyr/dt-bindings/display/gc9x01x.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+
+/ {
+	model = "ESP32-S3-Touch-LCD-1.28";
+	compatible = "waveshare,esp32-s3-touch-lcd-1.28";
+
+	aliases {
+		i2c-1 = &i2c1;
+		pwm-0 = &ledc0;
+		sw0 = &button0;
+		uart-0 = &uart0;
+		watchdog0 = &wdt0;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,display = &gc9a01;
+	};
+
+	/* Buttons */
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			gpios = <&gpio0 0 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "BOOT Button";
+			zephyr,code = <INPUT_KEY_0>;
+		};
+	};
+
+	/* Touch Controller */
+	lvgl_pointer {
+		compatible = "zephyr,lvgl-pointer-input";
+		input = <&cst816s>;
+	};
+
+	/* PWM */
+	pwmleds {
+		compatible = "pwm-leds";
+		backlight_pwm: backlight_pwm {
+			pwms = <&ledc0 0 PWM_MSEC(10) PWM_POLARITY_NORMAL>;
```

Comment:
you can also use `PWM_HZ`, may make more sense if you are matching with a spec in the datasheet given in frequency

Reply:
No specification here, this is a value I also seen on another board. I just copy/paste.

Reply:
ok, is 100Hz enough? no flicker at all? I see from https://files.waveshare.com/wiki/ESP32-S3-Touch-LCD-1.28/ESP32-S3-Touch-LCD-1.28-Sch.pdf that this drivers the LEDs directly, I'd suggest picking something faster, maybe 250Hz or something

Reply:
100Hz seems enough yes, but 250Hz is fine as well.